File: pi1_notebook/pi2_examples/Rice.java
// Auxiliary material for PI2 lecture, University of Mannheim
//
// Run with: javac Rice.java && java Rice
//
// Author: Rainer Gemulla
public class Rice {
  public static void main(String[] args) {
    long grains = 0;             // total number of grain on board

    long next = 1;               // grains placed on next square
    for (int i = 0; i<64; i++) {
      grains += next;
      next *= 2;
    }
    System.out.println("There are " + grains + " rice grains on the chessboard.");
    System.out.println("After adding one more rice grain, there are "
                       + (grains+1) + " rice grains on the chessboard.");
  }
}

File: pi1_notebook/pi2_examples/rice.c
// Auxiliary material for PI2 lecture, University of Mannheim
//
// Run with: clang rice.c && ./a.out
//
// Author: Rainer Gemulla

#include <stdio.h>
#include <stdint.h>

int main(int argc, char *argv[]) {
  uint64_t grains = 0;       // total number of grain on board

  uint64_t next = 1;           // grains placed on next square
  for (int i = 1; i<=64; i++) {
    grains += next;
    next *= 2;
  }
  printf("There are %lu rice grains on the board.\n", grains);
  printf("After adding one more rice grain, "
         "there are %lu rice grains on the board.\n",
         grains+1);

  return 0;
}
